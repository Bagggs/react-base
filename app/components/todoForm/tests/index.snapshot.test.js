import React from 'react';
import renderer from 'react-test-renderer';
import TodoForm from '../';

describe('TodoForm', () => {
  it('renders correctly', () => {
    const todoFormComponent = renderer.create(<TodoForm />);

    expect(todoFormComponent.toJSON()).toMatchSnapshot();
  });

  context('when name is invalid', () => {
    it('renders form with validation error', () => {
      const todoFormComponent = renderer.create(<TodoForm />);

      todoFormComponent.getInstance().setState({ name: 'qwe' });

      expect(todoFormComponent.toJSON()).toMatchSnapshot();
    });
  });
});
