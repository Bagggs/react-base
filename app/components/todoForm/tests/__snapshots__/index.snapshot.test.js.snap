// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoForm renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="modal-body"
  >
    <div
      className="form-group"
    >
      <label
        className="control-label"
        htmlFor="taskName"
      >
        Task name
        : 
        
      </label>
      <input
        className="form-control"
        id="taskName"
        name="name"
        onChange={[Function]}
        placeholder="Task name..."
        type="text"
        value=""
      />
    </div>
    <div
      className="modal-footer"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        type="submit"
      >
        Save
      </button>
    </div>
  </div>
</form>
`;

exports[`TodoForm when name is invalid renders form with validation error 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="modal-body"
  >
    <div
      className="form-group has-error"
    >
      <label
        className="control-label"
        htmlFor="taskName"
      >
        Task name
        : 
        qwe
      </label>
      <input
        className="form-control"
        id="taskName"
        name="name"
        onChange={[Function]}
        placeholder="Task name..."
        type="text"
        value="qwe"
      />
    </div>
    <div
      className="modal-footer"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        type="submit"
      >
        Save
      </button>
    </div>
  </div>
</form>
`;
